File: back/src/main/java/org/example/springbootapi/dto/node/NodeResponseDto.java
package org.example.springbootapi.dto.node;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.springbootapi.constant.NodeType;

import java.time.LocalDateTime;
import java.util.List;
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class NodeResponseDto {

    private Long id;
    private String name;
    private NodeType type;
    private Long size;
    private String mimeType;
    private String storageKey;
    private Long userId;
    private Long parentId;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: front/src/models/NodeResponseDto.ts
import type {NodeType} from "../pages/Finder.tsx";

export default interface NodeResponseDto {
    id: number;
    name: string;
    type: NodeType;
    size: number | null;
    mimeType: string | null;
    storageKey: string | null;
    userId: number;
    parentId: number | null;
    createdAt: string;
    updatedAt: string;
}